Assets/Scripts/EnemyBehaviour.cs

```diff
@@ -0,0 +1,180 @@
+
+        try {
+            this._CheckUseCase(state);
+        } catch(InvalidEnemyStateException ex) {
+            Debug.LogException(ex);
+        }
+
+        this._enemyBody = enemy.GetComponent<Rigidbody2D>();
+    }
+
+    /// <summary>
+    /// Sets the Enemy's velocity to zero on the current frame
+    /// </summary>
+    /// <returns> true </returns>
+    public override bool Act() {
+        this._enemyBody.velocity = Vector2.zero;
+        return true;
+    }
+}
+
+/// <summary>
+/// EnemyBehaviour in which Enemy increases velocity in direction of a Rigidbody2D target up to a maximum speed
+/// </summary>
+/// <remarks>
+/// Intended for use in aware and engaged states
+/// </remarks>
+public class ApproachTarget : EnemyBehaviour {
+    /// <summary>
+    /// Enemy SpriteRenderer
+    /// </summary>
+    protected SpriteRenderer _enemySprite;
+    /// <summary>
+    /// Target of the Enemy
+    /// </summary>
+    /// <remarks>
+    /// Usually the player
+    /// </remarks>
+    protected Rigidbody2D _target;
+    /// <summary>
+    /// Maximum speed at which enemy can move
+    /// </summary>
+    public float max_speed;
+    /// <summary>
+    /// Magnitude of enemy acceleration
+    /// </summary>
+    public float acceleration_rate;
+
+    /// <summary>
+    /// Constructs a new ApproachTarget behaviour
+    /// </summary>
+    /// <param name="enemy"> Enemy who will use the behaviour </param>
+    /// <param name="target"> Target that the Enemy will approach (e.g. the player character) </param>
+    /// <param name="max_speed"> Maximum speed at which Enemy can move </param>
+    /// <param name="acceleration_rate"> Magnitude of Enemy acceleration </param>
+    /// <param name="state"> ENEMY_STATE in which the behaviour will be used. Null to skip use-case checking </param>
+    public ApproachTarget(Enemy enemy, Rigidbody2D target, float max_speed, float acceleration_rate, ENEMY_STATE? state) {
+        this._use_case[(int)ENEMY_STATE.aware] = true;
+        this._use_case[(int)ENEMY_STATE.engaged] = true;
+
+        try {
+            this._CheckUseCase(state);
+        } catch(InvalidEnemyStateException ex) {
+            Debug.LogException(ex);
+        }
+
+        this._enemyBody = enemy.GetComponent<Rigidbody2D>();
+        this._enemySprite = enemy.GetComponent<SpriteRenderer>();
+        this._target = target;
+        this.max_speed = max_speed;
+        this.acceleration_rate = acceleration_rate;
+    }
+
+    /// <summary>
+    /// Increases Enemy's velocity in direction of _target
+    /// </summary>
+    /// <returns> true </returns>
+    public override bool Act() {
+        // Used to normalize updates across various frame-rates
+        float time_step = Time.fixedDeltaTime;
+
+        // Calculates change in velocity for current frame
+        Vector2 delta_v = this._target.position - this._enemyBody.position;
+        this._enemySprite.flipX = delta_v.x <= 0;
```

Comment:
Since different enemies way want different cosmetics even for the same action, I'm inclined to think we should care for those in the actual appropriate `Enemy` subclass, maybe with a similar `After<behavior>Animate` hook

Reply:
Is there any reason to use events to handle this instead of just making abstract AfterIdleAnimate, AfterAwareAnimate, and AfterEngagedAnimate methods that we implement in the subclass?

I thought at first that you wanted to use events for Animator compatibility, but the Animator uses a separate trigger system with boolean parameters that you set from inside C# scripts, so that's definitely not the motivating reason.

Reply:
I just learned what a hook is. I'm great at this.

Reply:
Alright this is now implemented, so this specific issue is resolved, but I'm going to raise a follow-up issue because of a choice I made in the implementation that I want your eyes on.